Tools/LambdaTestTool-v2/src/Amazon.Lambda.TestTool/Services/ApiGatewayRouteConfigService.cs

```diff
@@ -0,0 +1,135 @@
+        foreach (DictionaryEntry entry in environmentVariables)
+        {
+            var key = entry.Key.ToString();
+            if (key is null)
+                continue;
+            logger.LogDebug("Environment variables: {VariableName}", key);
+            if (!(key.Equals(Constants.LambdaConfigEnvironmentVariablePrefix) || 
+                key.StartsWith($"{Constants.LambdaConfigEnvironmentVariablePrefix}_")))
+            {
+                logger.LogDebug("Skipping environment variable: {VariableName}", key);
+                continue;
+            }
+
+            var jsonValue = entry.Value?.ToString()?.Trim();
+            logger.LogDebug("Environment variable value: {VariableValue}", jsonValue);
+            if (string.IsNullOrEmpty(jsonValue))
+                continue;
+
+            try
+            {
+                if (jsonValue.StartsWith('['))
+                {
+                    logger.LogDebug("Environment variable value starts with '['. Attempting to deserialize as a List.");
+                    var config = JsonSerializer.Deserialize<List<ApiGatewayRouteConfig>>(jsonValue);
+                    if (config != null)
+                    {
+                        _routeConfigs.AddRange(config);
+                        logger.LogDebug("Environment variable deserialized and added to the existing configuration.");
+                    }
+                    else
+                    {
+                        logger.LogDebug("Environment variable was not properly deserialized and will be skipped.");
+                    }
+                }
+                else
+                {
+                    logger.LogDebug("Environment variable value does not start with '['.");
+                    var config = JsonSerializer.Deserialize<ApiGatewayRouteConfig>(jsonValue);
+                    if (config != null)
+                    {
+                        _routeConfigs.Add(config);
+                        logger.LogDebug("Environment variable deserialized and added to the existing configuration.");
+                    }
+                    else
+                    {
+                        logger.LogDebug("Environment variable was not properly deserialized and will be skipped.");
+                    }
+                }
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine($"Error deserializing environment variable {key}: {ex.Message}");
+                logger.LogDebug("Error deserializing environment variable {Key}: {Message}", key, ex.Message);
+            }
+        }
+    }
+
+    /// <inheritdoc />
+    public ApiGatewayRouteConfig? GetRouteConfig(string httpMethod, string path)
+    {
+        foreach (var routeConfig in _routeConfigs)
+        {
+            _logger.LogDebug("Checking if '{Path}' matches '{Template}'.", path, routeConfig.Path);
+            var template = TemplateParser.Parse(routeConfig.Path);
```

Comment:
I don't see any tests for handling API Gateway's style of using `{proxy+}` to indicate wild card character. I really want to know if Microsoft's `TemplateParser` is handling that or if we need to roll our own solution. Have you done any side by side comparisons with API Gateway and this library to see if they resolve the same.

Reply:
TemplateParser supports a catch-all variable similar to the greedy variable in API gateway. I have updated the implementation to account for it as well as add sorting based on the route priority logic from API Gateway. I have added comments to explain everything in the code.